reconcile/test/aws_account_manager/test_aws_account_manager_integration.py

```diff
@@ -0,0 +1,239 @@
+) -> None:
+    intg.save_access_key(
+        "account", AWSAccessKey(AccessKeyId="access_key", SecretAccessKey="secret_key")
+    )
+    intg.secret_reader.vault_client.write.assert_called_once_with(  # type: ignore
+        secret={
+            "data": {
+                "aws_access_key_id": "access_key",
+                "aws_secret_access_key": "secret_key",
+            },
+            "path": "app-sre/creds/terraform/account/config",
+        },
+        decode_base64=False,
+    )
+
+
+def test_aws_account_manager_utils_integration_create_accounts(
+    aws_api: MagicMock,
+    reconciler: MagicMock,
+    merge_request_manager: MagicMock,
+    account_request: AWSAccountRequestV1,
+    intg: AwsAccountMgmtIntegration,
+) -> None:
+    reconciler.create_organization_account.return_value = None
+    intg.create_accounts(
+        aws_api,
+        reconciler,
+        merge_request_manager,
+        "account-template",
+        [account_request],
+    )
+    reconciler.create_organization_account.assert_called_once_with(
+        aws_api=aws_api,
+        name=account_request.name,
+        email=account_request.account_owner.email,
+    )
+    merge_request_manager.create_account_file.assert_not_called()
+
+
+def test_aws_account_manager_utils_integration_create_accounts_create_account_file(
+    aws_api: MagicMock,
+    reconciler: MagicMock,
+    merge_request_manager: MagicMock,
+    account_request: AWSAccountRequestV1,
+    intg: AwsAccountMgmtIntegration,
+) -> None:
+    reconciler.create_organization_account.return_value = "1111111111"
+    intg.create_accounts(
+        aws_api,
+        reconciler,
+        merge_request_manager,
+        "account-template - {{ uid }}",
+        [account_request],
+    )
+    reconciler.create_organization_account.assert_called_once_with(
+        aws_api=aws_api,
+        name=account_request.name,
+        email=account_request.account_owner.email,
+    )
+    merge_request_manager.create_account_file.assert_called_once_with(
+        account_request_file_path="/aws/data/request.yml",
+        account_tmpl_file_content="account-template - 1111111111",
+        account_tmpl_file_path="data/templating/collections/aws-account/data.yml",
+    )
+
+
+def test_aws_account_manager_utils_integration_reconcile_organization_accounts(
+    aws_api: MagicMock,
+    reconciler: MagicMock,
+    org_account: AWSAccountManaged,
+    intg: AwsAccountMgmtIntegration,
+) -> None:
+    intg.reconcile_account = MagicMock()  # type: ignore
+    intg.reconcile_organization_accounts(aws_api, reconciler, [org_account])
+    reconciler.reconcile_organization_account.assert_called_once_with(
+        aws_api=aws_api,
+        default_tags={},
+        enterprise_support=False,
+        name="jeanluc",
+        ou="/Root/alpha quadrant/uss enterprise/ncc-1701-d",
+        tags={"ship": "USS Enterprise"},
+        uid="111111111111",
+    )
+    intg.reconcile_account.assert_called_once()
```

Comment:
do you want to verify args for `reconcile_account` too? Or it's covered in another case.

Reply:
Not needed, there isn't so much logic here.